lib/readers.js

```diff
@@ -306,6 +307,14 @@ FrameReader.prototype.readMetadata = function (kind) {
   return meta;
 };
 
+// templates for derived error messages.
```

Comment:
In the jira ticket we discussed the possibility of creating constructors for each error type or using some kind of builder, but it didn't take too much extra code to add this inline so I opted to take the simple approach of adding the logic in the existing method, but can change it if requested.